### PR TITLE
Rename webjars-locator to web-dependency-locator

### PR DESCRIPTION
**Pull request by phillip-kruger**
As discussed, we rename (and relocate) the webjars-locator to something more generic, like web-dependency-locator, as now we support mvnpm and importmaps

### CODE REVIEW

**Review by gsmet — COMMENTED** [review]
Thanks! I added a few comments.

**Comment by ia3andy** [comment]
This needs to wait, we are discussing using this opportunity to move it to the quarkiverse. https://quarkusio.zulipchat.com/#narrow/stream/191168-core-team/topic/Quarkus.20Web/near/433256889

We are also discussing making it the way (opt-in via extension) to enable serving of mvnpm and webjars which is currently part of vert.x http.

File: extensions/web-dependency-locator/deployment/src/main/java/io/quarkus/webdependency/locator/deployment/ImportMapBuildItem.java [hunk 1]
```diff
@@ -1,4 +1,4 @@
-package io.quarkus.webjar.locator.deployment;
+package io.quarkus.webdependency.locator.deployment;
```
[the remark this answers is not in the document]
Comment (phillip-kruger) [reply extensions/web-dependency-locator/deployment/src/main/java/io/quarkus/webdependency/locator/deployment/ImportMapBuildItem.java:1]: This is a brand new Build Item. Not use anywhere but in the extension

**Comment by phillip-kruger** [comment]
@gsmet - please check again, I addressed your comments.
@ia3andy - It's not this extension that allows loading from META-INF. This extension only maps the versions path to the non-versioned path, and now also produce a importmap. 

We can probably move that logic to this extension. I would assume it's in vertx-http at the moment. That would not change if this extension is in Quarkiverse or here.

My suggestion is we merge here and then see if it will be possible to move that logic out of vertx-http and what the effort and impact would be.

**Comment by phillip-kruger** [comment]
Things that will break is swagger-ui and graphql-ui.

**Comment by ia3andy** [comment]
> Things that will break is swagger-ui and graphql-ui.

We might be able to keep the compat through builditems if we keep the logic in vertx http, but add the enablement through builditems.

This way the web-dependency extension can use this SPI for all the detected web dependencies and swagger-ui and graphql-ui can only add the required ones.

**Comment by gsmet** [comment]
Is there a reason why we would want to move it to the Quarkiverse?

It's useful and small and is already there so, except if we have a good reason to move it, I would keep it around.

**Comment by phillip-kruger** [comment]
I agree @gsmet

**Comment by ia3andy** [comment]
it was suggested by @cescoffier. I think it belongs more in quarkiverse and while renaming that might be a good occasion. I have no strong opinion on it.

**Comment by cescoffier** [comment]
The reason is just to reduce the size of the core and provide independent release cycles.

**Comment by phillip-kruger** [comment]
@gsmet you make a call. It would be good to get this into 3.10

**Comment by phillip-kruger** [comment]
Can this go into a patch release ? 3.10.1 ?

**Review by gsmet — APPROVED** [review]

**Comment by gsmet** [comment]
> Can this go into a patch release ? 3.10.1 ?

No we can't do that. It will have to wait for the next one.

**Pull request merged**